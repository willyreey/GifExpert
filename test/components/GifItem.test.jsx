const { render, screen, getByAltText } = require("@testing-library/react")
const { GifItem } = require("../../src/components/GifItem")

describe('Probando componente GifItem', () => { 
    
    const title = 'Goku'
    const url = 'https://gist.github.com/Klerith/ca7e57fae3c9ab92ad08baadc6c26177'

    
    
    test('Debe de hacer match con el snapshot', () => { 
        const { container } = render(<GifItem title={title} url={url}/>)

        expect(container).toMatchSnapshot()
     })

     test('Debe de mostrar la img con el texto y el ALT indicados', () => { 
        render(<GifItem title={title} url={url}/>)

        const {src, alt} = screen.getByRole('img')

        expect(src).toBe( url )
        expect(alt).toBe( title )

      })

      test('Debe de mostrar el titulo en el componente', () => { 
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByAltText(title)).toBeTruthy()
        
       })
 })