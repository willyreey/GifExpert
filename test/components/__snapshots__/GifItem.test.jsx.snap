// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando componente GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="https://gist.github.com/Klerith/ca7e57fae3c9ab92ad08baadc6c26177"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
